nomt/src/beatree/ops/update/leaf_updater.rs

```diff
@@ -460,10 +340,14 @@ impl LeafUpdater {
             pos += 1;
         }
 
+        // Use `pos - from` ops only if they create a node with a body size bigger than the target
+        // or accept a size below the target only if an item causes the node to transition
+        // from a body size below the target to overfull.
```

Comment:
note that this is actually impossible with our current leaf node. maybe we can simplify the code?

Our maximum value size is 1/3 of the page, so `1/2 + 1/3 < 1` and therefore `1/2 - x + 1/3 < 1`

Reply:
No, because the target is not always `1/2` of a leaf node. Look at the test `split_left_node_below_target` added in #643, it explicitly creates a scenario where the gauge does not trigger the bulk split, but instead a standard split occurs. The target of a standard split is `1/2 * gauge.body_size`, in this latter case, only the first x items are selected, and their body_size is below the target because the x + 1 items would be overfull.

This cannot be removed, as `try_build_leaves` would then have cases of wrong split failures, which is not expected by the `digest` method.

What you described is true for a target of `1/2` of the leaf node. That's why there is no check for the final body size to be bigger than `LEAF_MERGE_THRESHOLD`, because that scenario, as you explained, is impossible.

I will rearrange some comments to make these things clearer.

Reply:
However, I would like to have this double-checked, so I will ask for another review to make sure the math I did is correct.

Reply:
ahh, I see what you mean now. disregard! so for targets greater than 2/3 of a leaf node size (which could happen if we were splitting something >1.33x the max size) we could possibly jump from below the target to overfull